Tests/RandomAccessCollectionTests.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import XCTest
+
+class RandomAccessCollectionTests: XCTestCase {
+    func testEnumerateWithNext() {
+        let ary = [1,2,3]
+        for (current, next) in ary.enumerateWithNext() {
+            XCTAssert(current == 1)
+            if let next = next {
+                XCTAssert(next == 2)
+            }
+            break
+        }
+    }
+
+    func testEnumerateWithPrevious() {
+        let ary = [1,2,3]
+        for (previous, current) in ary.enumerateWithPrevious() {
+            if current == 1 { continue }
+            XCTAssert(current == 2)
+            if let previous = previous {
```

Comment:
But once we have that, what is the difference between `previous` and `next` methods? Just the order of the elements inside the tuple?